gazebo_ros/src/gazebo_ros_init.cpp

```diff
@@ -206,7 +210,12 @@ void GazeboRosInitPrivate::onPerformanceMetrics(
     msg_ros.sensors.push_back(sensor_msgs);
   }
 
-  performance_metrics_pub_->publish(msg_ros);
+  // Check if parameter was enabled
+  bool check_enable_performance_metrics ;
+  this->ros_node_->get_parameter("enable_performance_metrics", check_enable_performance_metrics);
+  if (check_enable_performance_metrics){
```

Comment:
IIUC, we don't need to execute the rest of the logic in `onPerformanceMetrics` above. I suggest we check this parameter at the beginning of the method and return early if the parameter value is `false`.

Reply:
Done